narwhal/primary/src/primary.rs

```diff
@@ -676,15 +687,26 @@ impl WorkerToPrimary for WorkerReceiverHandler {
         request: anemo::Request<WorkerOurBatchMessage>,
     ) -> Result<anemo::Response<()>, anemo::rpc::Status> {
         let message = request.into_body();
-        self.tx_our_digests
+        let (tx_ack, rx_ack) = tokio::sync::oneshot::channel();
+        let response = self
+            .tx_our_digests
             .send((
-                message.digest,
-                message.worker_id,
-                message.metadata.created_at,
+                (
+                    message.digest,
+                    message.worker_id,
+                    message.metadata.created_at,
+                ),
+                tx_ack,
             ))
             .await
             .map(|_| anemo::Response::new(()))
-            .map_err(|e| anemo::rpc::Status::internal(e.to_string()))
+            .map_err(|e| anemo::rpc::Status::internal(e.to_string()));
+
+        // TODO: if there is an error here, we should send an error back to the client
```

Comment:
Did it!

Reply:
Came back to take a second look and just flagging that the updates you've made are not pushed to github yet - maybe that's intentional but wanted to at least mention since you did reply to the comments. I will await you clicking the "re-request review" button :)